src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java

```diff
@@ -0,0 +1,499 @@
+		Transaction transaction = session.beginTransaction();
+
+		Query<Entity> query = session.createQuery("FROM EntityImpl entity WHERE entity.id = :id",
+				Entity.class)
+				.setParameter("id", id);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, null);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Entity getMappingEntity(String sqlQuery) {
+		if (sqlQuery == null || sqlQuery.length() == 0) {
+			return null;
+		}
+
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query query = session.createNativeQuery(sqlQuery);
+
+		List<?> entityIds = databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+
+		if (entityIds.size() != 1) {
+			return null;
+		}
+
+		return getEntity(Long.parseLong(entityIds.get(0).toString()));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Entity> getEntities(long parentId, long entityTypeId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Entity> query = session.createQuery("FROM EntityImpl entity "
+				+ "WHERE entity.typeId = :entityTypeId AND entity.parentId = :parentId",
+				Entity.class)
+				.setParameter("entityTypeId", entityTypeId)
+				.setParameter("parentId", parentId);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public EntityType getEntityType(long id) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<EntityType> query = session.createQuery("FROM EntityTypeImpl entityType WHERE entityType.id = :id",
+				EntityType.class)
+				.setParameter("id", id);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, null);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<EntityType> getEntityTypes() {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<EntityType> query = session.createQuery("FROM EntityTypeImpl entityType",
+				EntityType.class);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Event> getEvents(long entityOwnerId, long eventTypeId, int listStartIndex, int listEndIndex) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Event> query = session.createQuery("FROM EventImpl event "
+				+ "WHERE event.typeId = :eventTypeId AND event.entityId = :entityOwnerId",
+				Event.class)
+				.setParameter("eventTypeId", eventTypeId)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "eventTypeId" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)